### ISSUE TITLE
Godot Mono v3.1.1 crash when casting Node to Camera2D

### ISSUE THREAD

**Issue by PanosTrak** [godotengine/godot#30603]
**Godot version:**
Godot-mono stable v3.1.1
66baa3b

**OS/device including version:**
OS: Manjaro Linux
Kernel Version: 4.19.56-1
Mono JIT compiler version 5.20.1
GPU1: Intel Corporation HD Graphics 5500
GPU2: Advanced Micro Devices, Inc. [AMD/ATI] Opal XT [Radeon R7 M265/M365X/M465] (its the Radeon R7 M270 but it doesnt show up like this)

**Issue description:**
When i use 
`camera = (Camera2D)GetNode("camera");` 
i get this error
> drivers/unix/net_socket_posix.cpp:199 - Socket error: 104

**Minimal reproduction project:**
[Pirates.zip](https://github.com/godotengine/godot/files/3394304/Pirates.zip)

**Comment by KoBeWi** [comment]
I can run the project just fine, no errors.

**Comment by PanosTrak** [comment]
[2019_07_16 11.22.06 (1547).txt](https://github.com/godotengine/godot/files/3396049/2019_07_16.11.22.06.1547.txt)
[2019_07_16 11.21.57 (1504).txt](https://github.com/godotengine/godot/files/3396071/2019_07_16.11.21.57.1504.txt)

The pastebin bellow is the output to the terminal
https://pastebin.com/cdR05hzv

**Comment by akien-mga** [comment]
Stacktrace on thread 1:
```
=================================================================
    Managed Stacktrace:
=================================================================
      at <unknown> <0xffffffff>
      at Godot.Object:godot_icall_Object_ClassDB_get_method <0x000c2>
      at Godot.Camera2D:.cctor <0x0008b>
      at System.Object:runtime_invoke_void <0x000cb>
      at <unknown> <0xffffffff>
      at Godot.NativeCalls:godot_icall_1_514 <0x000e8>
      at Godot.Node:GetNode <0x000e3>
      at Godot.Node:GetNode <0x0008f>
      at PlayerScript:_Ready <0x000db>
      at System.Object:runtime_invoke_void__this__ <0x000ce>
=================================================================
```

**Comment by akien-mga** [comment]
Can you still reproduce the issue in the latest 3.2 builds?

I tried with 3.2 RC 1 on Linux and had no crash, the casted node is a `Camera2D` as expected.

**Comment by PanosTrak** [comment]
Well, i have a fresh install of linux and it seems to work okay even on 3.1.2.stable.mono.official, i think the problem was that the msbuild was outdated because the maintainer of this package on AUR made this package to depend on the msbuild-stable which was outdated, now its set to msbuild which seems to be fine!

**Comment by akien-mga** [comment]
Thanks, closing then.

**Comment by neikeq** [comment]
Looking at the log, this is the same as #29812, which was fixed by #31784 (the issue is still open because the fix is just a workaround).
